### PR TITLE
Fix GH-1875

### PR DESCRIPTION
**Pull request by mysticmind**
- Fixes GH-1875
- Add unit test

### CODE REVIEW

File: src/DocumentDbTests/Bugs/Bug_1875_duplicated_array_field_test.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using Shouldly;
+using Weasel.Core;
+using Xunit;
+
+namespace DocumentDbTests.Bugs
+{
+    public class Bug_1875_duplicated_array_field_test : BugIntegrationContext
+    {
+        [Fact]
+        public void query_on_duplicated_array_field_test()
```
Comment (oskardudycz) [inline src/DocumentDbTests/Bugs/Bug_1875_duplicated_array_field_test.cs:16]: **SUGGESTION:** @mysticmind, it'd be worth also checking the `Guid` array as in the original array. We had cases in the past that types like string, guids, dates were failing, while for numbers, it was fine, and we were lacking tests for them.

**Review by oskardudycz — APPROVED** [review]
Babu, you're on 🔥 😄

File: src/DocumentDbTests/Bugs/Bug_1875_duplicated_array_field_test.cs (hunk 1 above)
Comment (mysticmind) [reply src/DocumentDbTests/Bugs/Bug_1875_duplicated_array_field_test.cs:16]: @oskardudycz, the issue was reproducible for the unit test which I have added so we are good.

Comment (mysticmind) [reply src/DocumentDbTests/Bugs/Bug_1875_duplicated_array_field_test.cs:16]: There is no harm in adding another test using `Guid` array, will add it.

Comment (mysticmind) [reply src/DocumentDbTests/Bugs/Bug_1875_duplicated_array_field_test.cs]: Added another unit test for Guid array duplicate field.

Comment (oskardudycz) [reply src/DocumentDbTests/Bugs/Bug_1875_duplicated_array_field_test.cs]: Thank you, Babu! 🙂
